gstd/gstd_http.c

```diff
@@ -0,0 +1,423 @@
+      "Access-Control-Allow-Origin", "*");
+  soup_message_headers_append (msg->response_headers,
+      "Access-Control-Allow-Headers", "origin,range,content-type");
+  soup_message_headers_append (msg->response_headers,
+      "Access-Control-Allow-Methods", "PUT, GET, POST, DELETE");
+
+  if (msg->method == SOUP_METHOD_GET) {
+    do_request (server, msg, query, session, GET);
+  } else if (msg->method == SOUP_METHOD_POST) {
+    do_request (server, msg, query, session, POST);
+  } else if (msg->method == SOUP_METHOD_PUT) {
+    do_request (server, msg, query, session, PUT);
+  } else if (msg->method == SOUP_METHOD_DELETE) {
+    do_request (server, msg, query, session, DELETE);
+  } else {
+    soup_message_set_status (msg, SOUP_STATUS_NOT_IMPLEMENTED);
+  }
+
+}
+
+static GstdReturnCode
+gstd_http_start (GstdIpc * base, GstdSession * session)
+{
+  GError *error = NULL;
+  SoupServer *server = NULL;
+
+  g_return_val_if_fail (base, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  GstdHttp *self = GSTD_HTTP (base);
+  guint16 port = self->port;
+  gchar *address = self->address;
+
+  gstd_http_stop (base);
+
+  GST_DEBUG_OBJECT (self, "Getting HTTP address");
+  server = soup_server_new (SOUP_SERVER_SERVER_HEADER, "HTTP-Server", NULL);
+
+  GSocketAddress *sa;
+  sa = g_inet_socket_address_new_from_string (address, port);
+
+  soup_server_listen (server, sa, 0, &error);
+  if (error) {
+    goto noconnection;
+  }
+
+  soup_server_add_handler (server, NULL, server_callback, session, NULL);
+
+  return GSTD_EOK;
+
+noconnection:
+  {
+    GST_ERROR_OBJECT (self, "%s", error->message);
+    g_printerr ("%s\n", error->message);
+    g_error_free (error);
+    g_free (server);
+    return GSTD_NO_CONNECTION;
+  }
+}
+
+static gboolean
+gstd_http_init_get_option_group (GstdIpc * base, GOptionGroup ** group)
+{
+  g_return_val_if_fail (base, FALSE);
+  g_return_val_if_fail (group, FALSE);
+
+  GstdHttp *self = GSTD_HTTP (base);
+
+  GOptionEntry http_args[] = {
+    {"enable-http-protocol", 't', 0, G_OPTION_ARG_NONE, &base->enabled,
+        "Enable attach the server through given HTTP ports ", NULL}
+    ,
+    {"http-address", 'a', 0, G_OPTION_ARG_STRING, &self->address,
+          "Attach to the server through a given address (default 127.0.0.1)",
+        "http-address"}
+    ,
+    {"http-port", 'p', 0, G_OPTION_ARG_INT, &self->port,
+          "Attach to the server through a given port (default 5001)",
+        "http-port"}
+    ,
+    {NULL}
+  };
+  GST_DEBUG_OBJECT (self, "HTTP init group callback ");
+  *group = g_option_group_new ("gstd-http", ("HTTP Options"),
+      ("Show HTTP Options"), NULL, NULL);
+
+  g_option_group_add_entries (*group, http_args);
+  return TRUE;
+}
+
+static GstdReturnCode
+gstd_http_stop (GstdIpc * base)
+{
+  g_return_val_if_fail (base, GSTD_NULL_ARGUMENT);
+
+  GstdHttp *self = GSTD_HTTP (base);
+  GstdSession *session = base->session;
+
```

Comment:
GstdHttp has an: `gchar *address;` is it free at some point here?

Reply:
Yes, in the dispose method.

Reply:
Oh yes, you are right